types/react-router/v3/lib/IndexRoute.d.ts

```diff
@@ -12,17 +12,18 @@ import {
 type ComponentCallback = (err: any, component: RouteComponent) => any;
 type ComponentsCallback = (err: any, components: RouteComponents) => any;
 
-export interface IndexRouteProps {
+export interface IndexRouteProps<Props> {
```

Comment:
Try `<Props = any>`, otherwise this is a breaking change. Same for `RouteProps` and `PlainRoute`.